### ISSUE TITLE
[tvp] bestaudio gives ERROR: string index out of range

### ISSUE THREAD

**Issue by werid** [yt-dlp/yt-dlp#11356]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Poland

### Provide a description that is worded well enough to be understood

The audio download error's immediately when using grabbing bestaudio. 

According to `-F`, there's both m3u8 and ism protocol's, and the m3u8 works. 

```
[info] Available formats for 82976319:
ID                 EXT  RESOLUTION FPS CH │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
hls-audio0-English m3u8 audio only        │                  m3u8  │ audio only        unknown             [en] English
hls-audio0-Polski  m3u8 audio only        │                  m3u8  │ audio only        unknown             [pl] Polski
mss-English-2      isma audio only      2 │ ~  1.75MiB    2k ism   │ audio only        AACL         2k 48k [en]
mss-Polski-128     isma audio only      2 │ ~111.69MiB  128k ism   │ audio only        AACL       128k 48k [pl]
```

in case it's useful, i grabbed the sources with `--write-pages`, here's the relevant stuff (from what i can see):

```
<?xml version="1.0" encoding="utf-8"?>
<SmoothStreamingMedia MajorVersion="2" MinorVersion="0" Duration="66013140000">
  <StreamIndex Type="video" QualityLevels="7" Chunks="1651" Url="QualityLevels({bitrate})/Fragments(video={start time})">
    [...]
    <QualityLevel Index="6" Bitrate="4998336" FourCC="H264" MaxWidth="1920" MaxHeight="1080" CodecPrivateData="0000000167640029ACD940780227E5C04400000300040000
0300C83C60C6580000000168EBECB22C"></QualityLevel>
    <c n="0" d="40000000"></c>
    <c n="1" d="40000000"></c>
    <c n="2" d="40000000"></c>
    [...]
  <StreamIndex Type="audio" Name="Polski" Language="pl" QualityLevels="1" Chunks="1651" Url="QualityLevels({bitrate})/Fragments(audio={start time})">
    <QualityLevel Index="0" Bitrate="128000" FourCC="AACL" SamplingRate="48000" Channels="2" BitsPerSample="16" PacketSize="0" AudioTag="255" CodecPrivateData=
"119056E500"></QualityLevel>
    <c n="0" d="40062500"></c>
    <c n="1" d="40108125"></c>
    <c n="2" d="39894583"></c>
    [...]
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--vpn3', '-f', 'hls-audio0-Polski,mss-Polski-128', '-vU', 'https://sport.tvp.pl/82976319/eliminacje-me-2025-kobiet-bukareszt-rumunia-polska-mecz']
[debug] User config "/home/weird/.config/yt-dlp/config": ['--alias', '--vpn1', '--proxy http://192.168.0.11:8888', '--alias', '--vpn2', '--proxy http://192.168.0.12:8888', '--alias', '--vpn3', '--proxy http://192.168.0.13:8888', '--alias', '--vpn4', '--proxy http://192.168.0.14:8888', '--alias', '--vpn5', '--proxy http://192.168.0.15:8888', '--alias', '--vpn6', '--proxy http://192.168.0.16:8888', '--alias', '--vpn7', '--proxy http://192.168.0.17:8888', '--alias', '--vpn8', '--proxy http://192.168.0.18:8888', '--alias', '--vpn9', '--proxy http://192.168.0.19:8888']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.10.25.232902 from yt-dlp/yt-dlp-nightly-builds [57212a5f9] (zip)
[debug] Python 3.9.20 (CPython amd64 64bit) - FreeBSD-13.3-RELEASE-p5-amd64-64bit-ELF (OpenSSL 1.1.1w-freebsd  11 Sep 2023, libc 7)
[debug] exe versions: ffmpeg 6.1.2 (setts), ffprobe 6.1.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, requests-2.32.3, sqlite3-3.46.1, urllib3-2.1.0, websockets-13.0
[debug] Proxy map: {'all': 'http://192.168.0.13:8888'}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1839 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.10.25.232902 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.10.25.232902 from yt-dlp/yt-dlp-nightly-builds)
[tvp] Extracting URL: https://sport.tvp.pl/82976319/eliminacje-me-2025-kobiet-bukareszt-rumunia-polska-mecz
[tvp] 82976319: Downloading webpage
[tvp:embed] Extracting URL: tvp:82976319
[tvp:embed] 82976319: Downloading webpage
[tvp:embed] 82976319: Downloading ISM manifest
[tvp:embed] 82976319: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] 82976319: Downloading 2 format(s): hls-audio0-Polski, mss-Polski-128
[debug] Invoking hlsnative downloader on "https://sdt-thinx16-176.tvp.pl/token/video/vod/82976319/20241026/3240780296/09135832-6970-416c-9378-a29bd8a78b9c/video.ism/nv-hls-index-vod4-f1-a1.m3u8"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 1651
[download] Destination: Eliminacje ME 2025 kobiet, Bukareszt： Rumunia – Polska [MECZ] [82976319].m3u8
[download] 100% of  113.98MiB in 00:03:36 at 538.04KiB/s
[debug] Invoking ism downloader on "https://sdt-thinx16-176.tvp.pl/token/video/vod/82976319/20241026/3240780296/09135832-6970-416c-9378-a29bd8a78b9c/video.ism/manifest"
[ism] Total fragments: 1651
[download] Destination: Eliminacje ME 2025 kobiet, Bukareszt： Rumunia – Polska [MECZ] [82976319].isma
[download]   0.0% of ~ 210.27MiB at  107.78KiB/s ETA Unknown (frag 1/1651)ERROR: string index out of range
Traceback (most recent call last):
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 1625, in wrapper
    return func(self, *args, **kwargs)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 1781, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 1893, in process_ie_result
    return self.process_ie_result(
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 1840, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 3014, in process_video_result
    self.process_info(new_info)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 178, in wrapper
    return func(self, *args, **kwargs)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 3482, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/YoutubeDL.py", line 3202, in dl
    return fd.download(name, new_info, subtitle)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/downloader/common.py", line 466, in download
    ret = self.real_download(filename, info_dict)
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/downloader/ism.py", line 271, in real_download
    write_piff_header(ctx['dest_stream'], info_dict['_download_params'])
  File "/home/weird/bin/yt-dlp-nightly/yt_dlp/downloader/ism.py", line 87, in write_piff_header
    mdhd_payload += u16.pack(((ord(language[0]) - 0x60) << 10) | ((ord(language[1]) - 0x60) << 5) | (ord(language[2]) - 0x60))
IndexError: string index out of range
```

**Comment by bashonly** [comment]
Looks like ISM parsing fail. Could you test https://github.com/yt-dlp/yt-dlp/pull/11359 ?

I made it painless: `yt-dlp --update-to bashonly/yt-dlp@ism`

**Comment by bashonly** [comment]
I think I may have fixed it with #11359 (for real this time)

Could you test it please? `yt-dlp --update-to bashonly/yt-dlp@ism`

**Comment by werid** [comment]
thanks, now it downloads. 

```
[debug] Command-line config: ['-v', '--vpn3', '-f', 'ba', 'https://sport.tvp.pl/82976319/eliminacje-me-2025-kobiet-bukareszt-rumunia-polska-mecz']
[debug] User config "/home/weird/.config/yt-dlp/config": ['--alias', '--vpn1', '--proxy http://192.168.0.11:8888', '--alias', '--vpn2', '--proxy http://192.168.0.12:8888', '--alias', '--vpn3', '--proxy http://192.168.0.13:8888', '--alias', '--vpn4', '--proxy http://192.168.0.14:8888', '--alias', '--vpn5', '--proxy http://192.168.0.15:8888', '--alias', '--vpn6', '--proxy http://192.168.0.16:8888', '--alias', '--vpn7', '--proxy http://192.168.0.17:8888', '--alias', '--vpn8', '--proxy http://192.168.0.18:8888', '--alias', '--vpn9', '--proxy http://192.168.0.19:8888']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version bashonly/yt-dlp@ism build 2024.10.26.142909 [b05ece786] (zip)
[debug] Python 3.9.20 (CPython amd64 64bit) - FreeBSD-13.3-RELEASE-p5-amd64-64bit-ELF (OpenSSL 1.1.1w-freebsd  11 Sep 2023, libc 7)
[debug] exe versions: ffmpeg 6.1.2 (setts), ffprobe 6.1.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, requests-2.32.3, sqlite3-3.46.1, urllib3-2.1.0, websockets-13.0
[debug] Proxy map: {'all': 'http://192.168.0.13:8888'}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1837 extractors
[tvp] Extracting URL: https://sport.tvp.pl/82976319/eliminacje-me-2025-kobiet-bukareszt-rumunia-polska-mecz
[tvp] 82976319: Downloading webpage
[tvp:embed] Extracting URL: tvp:82976319
[tvp:embed] 82976319: Downloading webpage
[tvp:embed] 82976319: Downloading ISM manifest
[tvp:embed] 82976319: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] 82976319: Downloading 1 format(s): mss-Polski-128
[debug] Invoking ism downloader on "https://sdt-thinx12-172.tvp.pl/token/video/vod/82976319/20241026/3240780303/4d5f02eb-a163-4465-bb57-7708266466e0/video.ism/manifest"
[ism] Total fragments: 1651
[download] Destination: Eliminacje ME 2025 kobiet, Bukareszt： Rumunia – Polska [MECZ] [82976319].isma
[download] 100% of  104.07MiB in 00:03:48 at 465.76KiB/s
```
